lib/web/apiserver.go

```diff
@@ -5166,11 +5168,53 @@ func (h *Handler) authExportPublic(w http.ResponseWriter, r *http.Request, p htt
 		return
 	}
 
-	reader := strings.NewReader(authorities)
+	err = serveTextResponse(authorities, w, r)
+	if err != nil {
+		h.log.WithError(err).Error("Failed to encode and send response.")
+		http.Error(w, err.Error(), trace.ErrorToCode(err))
+	}
+}
+
+// Given an HTTP request with a textual response body, convert it to the requested response type.
+func serveTextResponse(responseBody string, w http.ResponseWriter, r *http.Request) error {
+	acceptHeaderValues := r.Header["Accept"]
+	// Add a default value in the event that either no value is sent, or no supported value is sent
+	// This must always return a plain text response to preserve backwards compatibility
+	acceptHeaderValues = append(acceptHeaderValues, "text/plain")
+
+	var responseContentType string
+	var reader io.ReadSeeker
+	for _, acceptHeaderValue := range acceptHeaderValues {
```

Comment:
This will work for:

```
Accept: content-type1
Accept: content-type2
```

But will it work for:
```
Accept: content-type1, content-type2
```
??

Reply:
I checked the source for `MIMEHeader.Values()` (which is what implements `req.Header().Values("Accept")` and it basically just does `req.Header()["Accept"]`, with no additional splitting.

That said, you're right about splitting the values. I misread the docs and thought this was handled automatically.